.claude/hooks/test/run.sh

```
@@ -0,0 +1,154 @@
+adeny() { printf '%s' "$1" | bash "$HERE/bash-deny.sh" >/dev/null 2>&1; echo $?; }
+mk() { printf '{"tool_name":"Bash","tool_input":{"command":"%s"},"cwd":"%s"}' "$1" "$PWD"; }
+chk "A denies --no-verify"          2 "$(adeny "$(mk 'git commit -m wip --no-verify')")"
+chk "A denies cargo fmt --all"      2 "$(adeny "$(mk 'cargo fmt --all')")"
+chk "A denies wrapped fmt --all"    2 "$(adeny "$(mk 'timeout 600 cargo fmt --all')")"
+chk "A denies git push --force"     2 "$(adeny "$(mk 'git push --force origin main')")"
+chk "A allows conventional commit"  0 "$(adeny "$(mk 'git commit -m \"feat(x): y\"')")"
+chk "A allows gh pr create"         0 "$(adeny "$(mk 'gh pr create --title \"Add X\"')")"
+chk "A allows grep literal"         0 "$(adeny "$(mk 'grep -rn \"TODO\" crates/')")"
+chk "A allows normal nextest"       0 "$(adeny "$(mk 'cargo nextest run -p nebula-engine')")"
+chk "A allows push no force"        0 "$(adeny "$(mk 'git push origin main')")"
+chk "A fail-open on subshell"       0 "$(adeny "$(mk 'cargo \$(echo test)')")"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)